presto-main/src/main/java/io/prestosql/dispatcher/DispatchManager.java

```diff
@@ -16,6 +16,7 @@
 import com.google.common.util.concurrent.AbstractFuture;
```

Comment:
Can you please add test to `io.prestosql.connector.system.TestKillQuery`. You can use MockConnector to block it on metadata calls so query will remain in `waiting for resources` state.

Reply:
@kokosing added.
It required some refactoring to `TestEventListeners` (separate commits)